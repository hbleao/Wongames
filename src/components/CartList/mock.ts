export const mockCartList = [
  {
    image: 'https://source.unsplash.com/user/williamjusten/151x70',
    title: 'Red Dead Redemption 2',
    price: 'R$ 215,00'
  },
  {
    image: 'https://source.unsplash.com/user/williamjusten/151x70',
    title: 'Borderlands 3',
    price: 'R$ 129,90'
  }
];
