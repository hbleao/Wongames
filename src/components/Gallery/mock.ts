export const mockGallery = [
  {
    src: '/img/games/cyberpunk-1.jpg',
    label: 'Gallery Image 1',
    alt: 'alt image 1'
  },
  {
    src: '/img/games/cyberpunk-2.jpg',
    label: 'Gallery Image 2',
    alt: 'alt image 2'
  },
  {
    src: '/img/games/cyberpunk-3.jpg',
    label: 'Gallery Image 3',
    alt: 'alt image 3'
  },
  {
    src: '/img/games/cyberpunk-4.jpg',
    label: 'Gallery Image 4',
    alt: 'alt image 4'
  },
  {
    src: '/img/games/cyberpunk-5.jpg',
    label: 'Gallery Image 5',
    alt: 'alt image 5'
  },
  {
    src: '/img/games/cyberpunk-6.jpg',
    label: 'Gallery Image 6',
    alt: 'alt image 6'
  }
];
