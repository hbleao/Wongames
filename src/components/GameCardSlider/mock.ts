import { GameCardProps } from 'components/GameCard/types';

export const mockGameCards: GameCardProps[] = [
  {
    title: 'Population 1',
    developer: 'Rockstar Games',
    image: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00',
    favorite: false
  },
  {
    title: 'Population 2',
    developer: 'Rockstar Games',
    image: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00',
    favorite: false
  },
  {
    title: 'Population 3',
    developer: 'Rockstar Games',
    image: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00',
    favorite: false
  },
  {
    title: 'Population 4',
    developer: 'Rockstar Games',
    image: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00',
    favorite: false
  },
  {
    title: 'Population 5',
    developer: 'Rockstar Games',
    image: 'https://source.unsplash.com/user/willianjusten/300x140',
    price: 'R$ 235,00',
    promotionalPrice: 'R$ 215,00',
    favorite: false
  }
];
