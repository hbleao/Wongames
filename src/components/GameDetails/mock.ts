import { Platform } from './types';

export const mockGameDetails = {
  developer: 'Different Tales',
  releaseDate: '2020-11-21T23:00:00',
  platforms: ['mac', 'linux', 'windows'] as Platform[],
  publisher: 'Walkabout',
  rating: 'BR0',
  genres: ['Role-playing', 'Adventure']
};
