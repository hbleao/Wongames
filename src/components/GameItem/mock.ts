import { GameItemProps } from './types';

export const mockGameItem: GameItemProps = {
  image: 'https://source.unsplash.com/user/willianjusten/151x70',
  title: 'Red dead Redemption 2',
  price: 'R$ 215,00',
  downloadLink: 'https://wowgames.com/games/download',
  paymentInfo: {
    flag: 'mastercard',
    image: '/img/master-card.png',
    number: '**** **** **** 4326',
    purchaseDate: 'Purchase made on 07/20/2020 at 20:32'
  }
};
