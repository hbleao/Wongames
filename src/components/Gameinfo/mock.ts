export const mockGameInfo = {
  title: 'Cyper Punk 2077',
  description:
    'Cyberpunk 2077 é um jogo eletrônico de RPG de ação desenvolvido pela CD Projekt Red e publicado pela CD Projekt. A história do jogo é ambientada em Night City, um mundo aberto situado no universo fictício de Cyberpunk.',
  price: '215,00'
};
