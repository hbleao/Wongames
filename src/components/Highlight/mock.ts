export const mockHighLight = {
  title: 'Read Dead is back!',
  subtitle: "Come see Jhon's new adventures",
  buttonLabel: 'Buy Now',
  backgroundImage: '/img/red-dead.png',
  buttonLink: '/rdr2'
};
