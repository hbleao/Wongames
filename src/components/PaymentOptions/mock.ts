export const mockPaymentOptions = [
  {
    flag: 'visa',
    image: '/img/visa.png',
    number: '**** **** **** 4378'
  },
  {
    flag: 'master',
    image: '/img/master-card.png',
    number: '**** **** **** 2490'
  }
];
