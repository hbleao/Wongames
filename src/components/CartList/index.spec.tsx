import React from 'react';
import { screen } from '@testing-library/react';

import { CartList } from '.';

import { renderWithTheme } from 'utils/tests/renderWithTheme';

import { mockCartList } from './mock';

const makeSut = () => {
  const sut = renderWithTheme(<CartList total="344,90" items={mockCartList} />);

  return {
    sut
  };
};

describe('CartList', () => {
  it('should be render a CartList', () => {
    const { sut } = makeSut();

    const headings = screen.getAllByRole('heading');
    expect(headings).toHaveLength(2);

    const total = screen.getByText(/total/i);
    expect(total).toBeInTheDocument();

    const price = screen.getByText(/344,90/i);
    expect(price).toBeInTheDocument();

    expect(sut).toMatchSnapshot();
  });
});
