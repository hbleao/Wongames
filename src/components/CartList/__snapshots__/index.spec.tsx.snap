// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartList should be render a CartList 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c3 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c6 {
  color: #FAFAFA;
  padding: 0.5rem 0.8rem;
  background-color: #3CD3C1;
  font-size: 1.4rem;
  font-weight: 600;
  border-radius: 0.4rem;
}

.c0 {
  background-color: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c7 {
  background-color: #EAEAEA;
  color: #030517;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  color: #F231A5;
  font-weight: 500;
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <img
                alt="Red Dead Redemption 2"
                src="https://source.unsplash.com/user/williamjusten/151x70"
              />
            </div>
            <div
              class="c4"
            >
              <h3
                class="c5"
              >
                Red Dead Redemption 2
              </h3>
              <div
                class="c6"
              >
                R$ 215,00
              </div>
            </div>
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <img
                alt="Borderlands 3"
                src="https://source.unsplash.com/user/williamjusten/151x70"
              />
            </div>
            <div
              class="c4"
            >
              <h3
                class="c5"
              >
                Borderlands 3
              </h3>
              <div
                class="c6"
              >
                R$ 129,90
              </div>
            </div>
          </div>
        </div>
        <div
          class="c7"
        >
          Total 
          <span
            class="c8"
          >
            344,90
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": .c1 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c3 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c6 {
  color: #FAFAFA;
  padding: 0.5rem 0.8rem;
  background-color: #3CD3C1;
  font-size: 1.4rem;
  font-weight: 600;
  border-radius: 0.4rem;
}

.c0 {
  background-color: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c7 {
  background-color: #EAEAEA;
  color: #030517;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  color: #F231A5;
  font-weight: 500;
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <img
              alt="Red Dead Redemption 2"
              src="https://source.unsplash.com/user/williamjusten/151x70"
            />
          </div>
          <div
            class="c4"
          >
            <h3
              class="c5"
            >
              Red Dead Redemption 2
            </h3>
            <div
              class="c6"
            >
              R$ 215,00
            </div>
          </div>
        </div>
      </div>
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <img
              alt="Borderlands 3"
              src="https://source.unsplash.com/user/williamjusten/151x70"
            />
          </div>
          <div
            class="c4"
          >
            <h3
              class="c5"
            >
              Borderlands 3
            </h3>
            <div
              class="c6"
            >
              R$ 129,90
            </div>
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        Total 
        <span
          class="c8"
        >
          344,90
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
